### DISCUSSION TITLE
Seeking out input for troubleshooting Sunbird installation

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#20] [Installation] [answered]
I have followed each and every step of this documentation:
http://docs.sunbird.org/1.9/developer-docs/installation/install_sbportal/

But when running I'm running this last command: **node server.js**
Getting following errors:

1. /home/ttpl-rt-58/sunbird-portal/src/app/node_modules/cassandra-driver/lib/client-options.js:94
    throw new TypeError('Contacts points are not defined.');

2. Cannot read property 'clientId' of undefined\n    at Object.<anonymous> (/home/ttpl-rt-58/sunbird-portal/src/app/helpers/userService.js:6:46)\n    at Module._compile (internal/modules/cjs/loader.js:1158:30)\n    at Object.Module._extensions..js

**Comment by luckynvdu** [comment]
@sharathkashyap @rajeevsathish Request to help here?

**Comment by rajeevsathish** [comment]
@nidhikachhawa24  can you please let us know which version of sunbird you are using and also what is the node version you have installed in your local machine

Cc: @luckynvdu @sharathkashyap

**Reply by nidhikachhawa24** [reply]
Hi @rajeevsathish 

I'm using
    "@project-sunbird/discussion-service": "0.0.6",
    "@project-sunbird/ext-framework-server": "0.1.28",
    "@project-sunbird/form-service": "0.0.14",
    "@project-sunbird/logger": "^0.0.9",
    "@project-sunbird/review-comment": "0.0.8",

Node version is v12.16.1

**Reply by nidhikachhawa24** [reply]
Hello @rajeevsathish,
I wrongly interpreted about the version: I came to know that I was doing setup from the master branch.

Now I have changed my branch to version 4.9.0 and starting the installation

cc: @Baljeetsingh

**Reply by nidhikachhawa24** [reply]
Hi @rajeevsathish 

Getting this error on running **$ node server.js** command

{"level":"error","message":{"error":{},"msg":"cassandraUtil : Error in getReplicationStrategy"},"pdata":{"pid":"sunbird-portal-backend","ver":"4.9.0"},"timestamp":"2022-06-10 11:42:54"}
{"date":"Fri Jun 10 2022 11:42:56 GMT+0530 (India Standard Time)","error":{},"exception":true,"level":"error","message":"uncaughtException: Cannot read property 'clientId' of undefined\nTypeError: Cannot read property 'clientId' of undefined\n    at Object.<anonymous> (/home/ttpl-rt-58/sunbird-portal/src/app/helpers/userService.js:6:46)\n    at Module._compile (internal/modules/cjs/loader.js:1158:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)\n    at Module.load (internal/modules/cjs/loader.js:1002:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:901:14)\n    at Module.require (internal/modules/cjs/loader.js:1044:19)\n    at require (internal/modules/cjs/helpers.js:77:18)\n    at Object.<anonymous> (/home/ttpl-rt-58/sunbird-portal/src/app/server.js:45:21)\n    at Module._compile (internal/modules/cjs/loader.js:1158:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)\n    at Module.load (internal/modules/cjs/loader.js:1002:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:901:14)\n    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)\n    at internal/main/run_main_module.js:18:47","os":{"loadavg":[1.40283203125,0.99267578125,1.068359375],"uptime":64776},"pdata":{"pid":"sunbird-portal-backend","ver":"4.9.0"},"process":{"argv":["/home/ttpl-rt-58/.nvm/versions/node/v12.16.1/bin/node","/home/ttpl-rt-58/sunbird-portal/src/app/server.js"],"cwd":"/home/ttpl-rt-58/sunbird-portal/src/app","execPath":"/home/ttpl-rt-58/.nvm/versions/node/v12.16.1/bin/node","gid":1001,"memoryUsage":{"external":3120728,"heapTotal":90173440,"heapUsed":59080560,"rss":137129984},"pid":21615,"uid":1001,"version":"v12.16.1"},"stack":"TypeError: Cannot read property 'clientId' of undefined\n    at Object.<anonymous> (/home/ttpl-rt-58/sunbird-portal/src/app/helpers/userService.js:6:46)\n    at Module._compile (internal/modules/cjs/loader.js:1158:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)\n    at Module.load (internal/modules/cjs/loader.js:1002:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:901:14)\n    at Module.require (internal/modules/cjs/loader.js:1044:19)\n    at require (internal/modules/cjs/helpers.js:77:18)\n    at Object.<anonymous> (/home/ttpl-rt-58/sunbird-portal/src/app/server.js:45:21)\n    at Module._compile (internal/modules/cjs/loader.js:1158:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)\n    at Module.load (internal/modules/cjs/loader.js:1002:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:901:14)\n    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)\n    at internal/main/run_main_module.js:18:47","timestamp":"2022-06-10 11:42:56","trace":[{"column":46,"file":"/home/ttpl-rt-58/sunbird-portal/src/app/helpers/userService.js","function":null,"line":6,"method":null,"native":false},{"column":30,"file":"internal/modules/cjs/loader.js","function":"Module._compile","line":1158,"method":"_compile","native":false},{"column":10,"file":"internal/modules/cjs/loader.js","function":"Module._extensions..js","line":1178,"method":".js","native":false},{"column":32,"file":"internal/modules/cjs/loader.js","function":"Module.load","line":1002,"method":"load","native":false},{"column":14,"file":"internal/modules/cjs/loader.js","function":"Module._load","line":901,"method":"_load","native":false},{"column":19,"file":"internal/modules/cjs/loader.js","function":"Module.require","line":1044,"method":"require","native":false},{"column":18,"file":"internal/modules/cjs/helpers.js","function":"require","line":77,"method":null,"native":false},{"column":21,"file":"/home/ttpl-rt-58/sunbird-portal/src/app/server.js","function":null,"line":45,"method":null,"native":false},{"column":30,"file":"internal/modules/cjs/loader.js","function":"Module._compile","line":1158,"method":"_compile","native":false},{"column":10,"file":"internal/modules/cjs/loader.js","function":"Module._extensions..js","line":1178,"method":".js","native":false},{"column":32,"file":"internal/modules/cjs/loader.js","function":"Module.load","line":1002,"method":"load","native":false},{"column":14,"file":"internal/modules/cjs/loader.js","function":"Module._load","line":901,"method":"_load","native":false},{"column":12,"file":"internal/modules/run_main.js","function":"Function.executeUserEntryPoint [as runMain]","line":74,"method":"executeUserEntryPoint [as runMain]","native":false},{"column":47,"file":"internal/main/run_main_module.js","function":null,"line":18,"method":null,"native":false}]}

**Reply by rajeevsathish** [reply]
@nidhikachhawa24 I hope now you are using node version more >= 14+ and also this error is that some of the env variables need to have values 
like 
envHelper.KEYCLOAK_GOOGLE_CLIENT.clientId
this is specific to your implementation

**Comment by luckynvdu** [comment, marked as answer]
@nidhikachhawa24 is your issue resolved with the proposed solution? if yes then please mark it as answered.

@rajeevsathish @sharathkashyap
